src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,193 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
+    public Node<T> nodeByIndex(int index) {
+        if (index < size / 2) {
+            newNode = first;
+            for (int i = 0; i < index; i++) {
+                newNode = newNode.next;
+            }
+            return newNode;
+        } else {
+            newNode = last;
+            for (int i = size - 1; i > index; i--) {
+                newNode = newNode.prev;
+            }
+            return newNode;
+        }
+    }
+
+    public void addFirst(T value) {
+        newNode = new Node(value);
+        first = last = newNode;
+    }
+
+    public void addFront(T value) {
+        newNode = new Node(value);
+        first.prev = newNode;
+        newNode.next = first;
+        newNode.prev = null;
+        first = newNode;
+    }
+
+    public void addBack(T value) {
+        newNode = new Node(value);
```

Comment:
```suggestion
        newNode = new Node(last, value, null);
        last.next = newNode;
        last = newNode;
```

Reply:
I tried to make realization with this constructor for the first time, but then I thought it will be simpler to write bounds  outside of constructor... and forgot to delete it